### PR TITLE
DRAFT: Go Support in Pulumi 2.0

### PR DESCRIPTION
**Pull request by spara**

### CODE REVIEW

File: content/blog/go-support-pulumi-2-0/index.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+date: 2020-05-06T13:10:24-05:00
+meta_desc: "Go is fully supported in Pulumi 2.0"
+meta_image: meta.png
+authors:
+    - evan-boyle
+tags:
+    - Go
+---
+
+We use Go to build Pulumi, making everyone on our engineering team a Gopher. Our open-source pulumi engine and SaaS back is written in Go, which has quickly become "the language of cloud." [Pulumi 2.0](https://www.pulumi.com/blog/pulumi-2-0/) affirms our commitment to Go by welcoming it to the family of supported infrastructure as code languages.
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:12]:
```suggestion
Over the last 10 years, Go has quickly become the "language of the cloud" for building application servers and services that run in and on today's cloud platforms. With [Pulumi 2.0](https://www.pulumi.com/blog/pulumi-2-0/, Go can also be used to manage and provision modern infrastructure as well.  Across any cloud (AWS, Azure, GCP, Kubernetes and [more than 50 others](https://www.pulumi.com/docs/reference/pkg/)!) and across a variety of workloads (containers, serverless, kubernetes, core infrastructure and more), you can now use the rich software engineering strengths of the Go language and ecosystem to manage your cloud infrastructure.  The [Pulumi open source project](https://github.com/pulumi/pulumi) itself has been built on Go from day 1, and so we're really excited to bring full Go support for cloud infrastructure as code to the same language ecosystem that Pulumi itself has been part of.
```

File: content/blog/go-support-pulumi-2-0/index.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+<!--more-->
+
+## What is Pulumi?
+
+Pulumi's approach to modern Infrastructure as Code summarized in four points.
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:18]:
```suggestion
Pulumi is a modern infrastructure as code tool for the cloud.  It is built around four key ideas.
```

File: content/blog/go-support-pulumi-2-0/index.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+- Programming Languages. Offer the best languages and tools to declare infrastructure and gain best-in-class productivity and engineering capabilities, leverage world-class language and tooling innovation, and connect to existing communities and ecosystems.
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:20]:
```suggestion
- Programming Languages. Pulumi offers best-in-class productivity and software engineering capabilities by allowing you to use existing rich programming languages like JavaScript, Python, .NET and of course Go.  Pulumi combines this with still getting the reliability of a desired-state infrastructure as code.  The best of both worlds! ```

File: content/blog/go-support-pulumi-2-0/index.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+- Multi-Cloud. Give operators the tools to build consistent workflows across any cloud(s). Whether public, private, or hybrid — you can use the full breadth and depth of any cloud services.
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:22]:
```suggestion
- Multi-Cloud. Pulumi enables you to build consistent workflows across any cloud(s). Whether public, private, or hybrid — you can use the full breadth and depth of any cloud services.
```

File: content/blog/go-support-pulumi-2-0/index.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+- Modern Application Architectures. Pulumi fully supports legacy workloads, but it is designed for the world of modern containers, serverless, and cloud-native architectures. Build the architecture that works for you by mixing and matching infrastructure components.
+
+- Cloud Engineering Teams. Pulumi is for the entire team; everyone ranging from developers, infrastructure and operations engineers, and security engineers can use Pulumi.  Together, they can tackle cloud solutions together without technology or team silos.
+
+## Giving you superpowers
+
+Pulumi gives you the productivity superpowers of real programming languages. But what can you do with them?
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:30]:
```suggestion
Pulumi gives you the software engineering superpowers of the Go programming language for working with cloud infrastructure. But what can you do with them?  In this post, we'll walk through a few vignettes of how Pulumi can be used to work with serverless, containers and Kubernetes workloads across GCP, Azure, AWS and self-managed clouds.
```

File: content/blog/go-support-pulumi-2-0/index.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+			args,
+			pulumi.DependsOn([]pulumi.Resource{logPolicy}),
+		)
+		if err != nil {
+			return err
+		}
+
+		// Export the lambda ARN.
+		ctx.Export("lambda", function.Arn)
+
```
Comment (lukehoban) [inline content/blog/go-support-pulumi-2-0/index.md:57]: Does this indentation render well in the site?   We may need to undent all of these.

File: content/blog/go-support-pulumi-2-0/index.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+	Replicas          pulumi.IntPtrInput
+}
+
+func NewServiceDeployment(...){...}
+```
+
+This allows us to simplify boilerplate and design strongly-typed APIs for our Kubernetes infrastructure (in this case, combining a `Service` and a `Deployment` into a single, simple, API with nice defaults for our use case):
+
+```go
+                       _, err := NewServiceDeployment(ctx, "redis-leader", &ServiceDeploymentArgs{
```
Comment (EvanBoyle) [inline content/blog/go-support-pulumi-2-0/index.md:198]: The indentation here is still a little off if you look at the rendered site.

File: content/blog/go-support-pulumi-2-0/index.md (hunk 6 above)
Comment (spara) [reply content/blog/go-support-pulumi-2-0/index.md:57]: example of code rendering 

Screen Shot 2020-05-13 at 5 51 04 PM

File: content/blog/go-support-pulumi-2-0/index.md (hunk 7 above)
Comment (spara) [reply content/blog/go-support-pulumi-2-0/index.md:198]: The rendering for go is more indented than for other languages, but otherwise, it looks ok.

**Comment by spara** [comment]
> +1 to @lukehoban's comments, but otherwise looks good to me! When are we planning on pushing this out?

next Tuesday on the 19th

File: content/blog/go-support-pulumi-2-0/index.md (hunk 7 above)
Comment (EvanBoyle) [reply content/blog/go-support-pulumi-2-0/index.md:198]: Screen Shot 2020-05-13 at 3 57 43 PM

In this instance, there may just be some extra spaces inserted. The `_, err` should have the same indentation as the `})` three lines below

**Pull request merged**